### PR TITLE
feat(payout_link): return total_count in filtered payouts list API response

### PR DESCRIPTION
**Pull request by kashif-m**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
This PR refactors the payouts filtered list's API response to include `total_count` which specifies the total number of payout entries available for the given constraints.

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?
1. Create payouts (create multiple for listing based on different constraints)
<details>
  <summary>2. Filter payouts</summary>
    curl --location 'http://localhost:8080/payouts/list' \
        --header 'Content-Type: application/json' \
        --header 'api-key: dev_cp78v8fT2C03KkEClI9aBMrqKF6N0SBLyUjasXtUQMbLTasfR2LJwjg0LlNHY060' \
        --data '{}'

image
</details>

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by lsampras — APPROVED** [review]

File: crates/router/src/core/payouts.rs [hunk 1]
```diff
@@ -897,10 +897,28 @@ pub async fn payouts_filtered_list_core(
     .map(ForeignFrom::foreign_from)
     .collect();
 
+    let total_count = db
+        .get_total_count_of_filtered_payouts(
+            merchant_account.get_id(),
+            filters.connector.clone(),
+            filters.currency.clone(),
+            filters.status.clone(),
+            filters.payout_method.clone(),
```
Comment (hrithikesh026) [inline crates/router/src/core/payouts.rs:906]: what about other parameters in `filter`. Are not supposed to apply those filters?

Comment (kashif-m) [reply crates/router/src/core/payouts.rs:906]: No, this is only for the payout specific filters

cc: @sagarnaikjuspay can you confirm?

**Review by hrithikesh026 — APPROVED** [review]
LGTM

File: api-reference/openapi_spec.json [hunk 2]
```diff
@@ -4345,7 +4345,7 @@
             "content": {
               "application/json": {
                 "schema": {
-                  "$ref": "#/components/schemas/PayoutListResponse"
+                  "$ref": "#/components/schemas/PayoutListResponseV2"
```
Comment (SanchithHegde) [inline api-reference/openapi_spec.json:4348]: Why is this included in the v1 OpenAPI specification? 

1. Is it intentional, or was it meant to be only updated in the v2 OpenAPI specification?
2. If it was intentional, can't we make changes in the `PayoutListResponse` struct instead?

File: crates/storage_impl/src/payouts/payouts.rs [hunk 3]
```diff
@@ -701,6 +725,124 @@ impl<T: DatabaseStore> PayoutsInterface for crate::RouterStore<T> {
...
+    #[instrument(skip_all)]
+    async fn get_total_count_of_filtered_payouts(
+        &self,
+        merchant_id: &common_utils::id_type::MerchantId,
+        active_payout_ids: &[String],
+        connector: Option<Vec<PayoutConnectors>>,
+        currency: Option<Vec<storage_enums::Currency>>,
+        status: Option<Vec<storage_enums::PayoutStatus>>,
+        payout_type: Option<Vec<storage_enums::PayoutType>>,
+        _storage_scheme: MerchantStorageScheme,
```
Comment (SanchithHegde) [inline crates/storage_impl/src/payouts/payouts.rs:739]: We can remove `storage_scheme` parameter if it is not being used anywhere, in both the `get_total_count_of_filtered_payouts()` and `filter_active_payout_ids_by_constraints()` methods.

File: crates/hyperswitch_domain_models/src/payouts/payouts.rs [hunk 4]
```diff
@@ -61,10 +61,29 @@ pub trait PayoutsInterface {
     #[cfg(feature = "olap")]
     async fn filter_payouts_by_time_range_constraints(
         &self,
-        merchant_id: &id_type::MerchantId,
-        time_range: &api_models::payments::TimeRange,
-        storage_scheme: MerchantStorageScheme,
+        _merchant_id: &id_type::MerchantId,
+        _time_range: &api_models::payments::TimeRange,
+        _storage_scheme: MerchantStorageScheme,
```
Comment (SanchithHegde) [inline crates/hyperswitch_domain_models/src/payouts/payouts.rs:66]: @dracarys18 Do we need the `storage_scheme` as a parameter for find / filter DB calls?

**Review by SanchithHegde — APPROVED** [review]

**Review by hrithikesh026 — APPROVED** [review]

**Review by lsampras — APPROVED** [review]
